SoftwareDesign.md

```diff
@@ -0,0 +1,11 @@
+I really enjoyed utilizing the pipenc package to install and create a development envrironment. 
+I have always been very intrigued by working on projects like this as I think its a little easier when 
+you are handling so many different packages. I think this would be useful in high production environments especially 
+when you are trying to add new funsctionalty to something and don't want to break what is already 
+working. I really like Docker whcih does something similar and is a good tool as well for deployment and
+ this reminded me of it a bit.
+ 
+An anti-pattern I encountered, and one that I have been doing a lot, is not commiting often and not
```

Comment:
I definitely agree that committing your code often is super important. It not only helps reverse changes in code simply, it also helps you to better visualize milestones and protects you in case something goes wrong. Definitely try to commit more often!